bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/RequestValidationService.kt

```diff
@@ -537,14 +566,39 @@ class RequestValidationService(
         val request: RequestWithKey
     )
 
-    fun <TError : ErrorCode, TRequest : RequestWithKey> mergeErrorMaps(
-        firstMap: Map<TRequest, Collection<ErrorInfo<TError>>>,
-        secondMap: Map<TRequest, Collection<ErrorInfo<TError>>>
-    ): Map<TRequest, Collection<ErrorInfo<TError>>> =
-        (firstMap.keys + secondMap.keys)
-            .associateWith {
-                (firstMap[it] ?: emptyList()) + (secondMap[it] ?: emptyList())
-            }
+    data class LegalEntityUpdateBridge(
+
+        val legalEntity: IBaseLegalEntityDto,
+        val request: RequestWithKey,
+        val bpnl: String
+    )
+
+    data class AddressBridge(
+
+        val request: RequestWithKey,
+        val address: IBaseLogisticAddressDto,
+        val bpn: String?
+    )
+
+    data class SiteCreateBridge(
+
+        val request: RequestWithKey,
+        val bpnlParent: String
+    )
+
+    data class SiteUpdateBridge(
+
+        val request: RequestWithKey,
+        val bpns: String
+    )
+
+    data class ValidatorMessages<out ERROR : ErrorCode>(
+
+        val regionNotFound: ERROR,
+        val identifierNotFound: ERROR,
+        val duplicateIdentifier: ERROR
+    )
+
 
     companion object {
         fun <TError : ErrorCode, TRequest : RequestWithKey> mergeErrorMaps(
```

Comment:
We could generalize this function and move it to `common.CollectionUtils`:

```suggestion
fun <KEY, VALUE> mergeMaps(
    vararg inputMaps: Map<out KEY, Collection<VALUE>>
): Map<KEY, Collection<VALUE>> {
    return inputMaps
        .map { inputMap ->
            inputMap.flatMap { (key, values) -> values.map { value -> Pair(key, value) } }
        }
        .reduce { entries1, entries2 -> entries1 + entries2 }
        .groupBy({ it.first }, { it.second })
}
```